openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -58,35 +59,35 @@ def get_completed_programs(student):
         student (User): Representing the student whose completed programs to check for.
 
     Returns:
-        list of program ids
+        list of program UUIDs
 
     """
-    meter = ProgramProgressMeter(student)
 
-    return meter.completed_programs
+    programs = [program['uuid'] for program in get_programs(student)]
```

Comment:
This is wrong. Take a look at what `ProgramProgressMeter` is doing and compare it to what `get_programs` does. You need to use `ProgramProgressMeter` to compute the set of programs the student has completed. I can see how `get_programs` taking a `user` argument is deceptive: that argument is used to authenticate requests to the catalog service, but the catalog service can't do filter by a user's completed programs.

If you left this as is, the task would think every student had completed every program, and they'd then be issued certs in all programs.